Engine/src/Engine/JobQueue.cpp

```diff
@@ -0,0 +1,54 @@
+#include "Log.h"
+#include "JobQueue.h"
+
+using namespace engine_concurrent;
+
+JobQueue* JobQueue::instance = nullptr;
+std::mutex JobQueue::instanceMutex;
+std::mutex JobQueue::queueMutex;
+
+JobQueue* JobQueue::getInstance()
+{
+	std::unique_lock lock(instanceMutex);
+	if (instance == nullptr)
+	{
+		instance = new JobQueue();
+	}
+	return instance;
+}
+
+JobQueue::JobQueue() : m_jobList(new std::list<ThreadJob*>()) {}
+JobQueue::~JobQueue() { delete m_jobList; }
+
+void JobQueue::postJob(ThreadJob* job)
+{
+	std::unique_lock lock(queueMutex);
+	//std::condition_variable
+	if (m_jobList->empty())
+	{
+		m_jobList->push_back(job);
+		return;
+	}
+	// Iterate over the job list
+	auto it = m_jobList->begin();
+	for (; (it != m_jobList->end() // If we're not at the end of JobList,

```

Comment:
How does this behave if the jobList is already full? would it not overwrite the last job in the list?